v2/internal/genericarmclient/policy_register_rp.go

```diff
@@ -64,11 +64,16 @@ func (r *rpRegistrationPolicy) Do(req *azpolicy.Request) (*http.Response, error)
 		return resp, err
 	}
 	var reqErr requestError
-	if err = runtime.UnmarshalAsJSON(resp, &reqErr); err != nil {
-		return resp, err
+	if unmarshalErr := runtime.UnmarshalAsJSON(resp, &reqErr); unmarshalErr != nil {
+		return resp, unmarshalErr
 	}
 	if reqErr.ServiceError == nil {
-		return resp, errors.New("missing error information")
+		// This is unexpected and likely due to Azure RPs failing to adhere to the expected
+		// error shape contract. The good news is that if this happens we know this isn't
+		// an unregistered RP because that API will have the right shape. We just return
+		// the resp/err directly in this case to let the caller deal with whatever the problem
+		// was as we know it wasn't unregistered RP.
```

Comment:
I don't understand this - it appears to be saying that we know this is a registered RP because those have the right shape, but isn't the problem here due to the result having the wrong shape?

Reply:
What I'm trying to say is, the error for unregistered RP will have the correct shape. So if we end up in this if (incorrect shape), it had to come from the RP, which means its registered, which means we can just return the error as-is we don't need to try to parse/process it.